### PR TITLE
multi: add deadline for first connection of new LNC conn

### PR DESCRIPTION
**Pull request by ellemouton**
In this PR, we add a deadline for the initial connection of an LNC
connection. So with this, the user is forced to use their pairing phrase
within a certain time frame. After this initial connection, future
connections are made with the second handshake version meaning that the
pairing phrase is rendered useless. By adding a time limit to the time
in which a user can use their pairing phrase, we reduce the risk created
by the users pairing phrase being leaked. The default time limit is set
to 10 minutes but can be customsed with the new `firstlncconndeadline`
flag.

The first commit in this PR adds a new `CreatedAt` field to `Session` so that 
we can have an absolute deadline.

Depends on https://github.com/lightninglabs/lightning-node-connect/pull/53

Fixes https://github.com/lightninglabs/lightning-terminal/issues/406

### CODE REVIEW

**Review by jamaljsr — APPROVED** [review]
tACK LGTM.. amazing turn around time 🥇

File: session/tlv.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ const (
 	typeLocalPrivateKey tlv.Type = 10
 	typeRemotePublicKey tlv.Type = 11
 	typeMacaroonRecipe  tlv.Type = 12
+	typeCreatedAt       tlv.Type = 13
```
Comment (guggero) [inline session/tlv.go:27]: Yayy, TLV for the win :star_struck:

File: session_rpcserver.go [hunk 2]
```diff
@@ -214,15 +215,92 @@ func (s *sessionRpcServer) resumeSession(sess *session.Session) error {
...
+			s.cfg.firstConnectionDeadline,
+		)
+
+		if deadlineDuration.Before(time.Now()) {
+			log.Debugf("Deadline for session %s has already "+
+				"passed. Revoking session", pubKeyBytes)
+
+			err = s.db.RevokeSession(pubKey)
+			if err != nil {
+				log.Debugf("error revoking session: "+"%v", err)
```
Comment (guggero) [inline session_rpcserver.go:241]: Any reason for not returning the error here? Or was this intended to be a `return fmt.Errorf()`? Also, string concatenation artifacts.

File: config.go [hunk 3]
```diff
@@ -302,21 +304,22 @@ func defaultConfig() *Config {
...
+		MacaroonPath:         DefaultMacaroonPath,
+		ConfigFile:           defaultConfigFile,
+		FaradayMode:          defaultFaradayMode,
+		Faraday:              &faradayDefaultConfig,
+		faradayRpcConfig:     &frdrpcserver.Config{},
+		LoopMode:             defaultLoopMode,
+		Loop:                 &loopDefaultConfig,
+		PoolMode:             defaultPoolMode,
+		Pool:                 &poolDefaultConfig,
+		FirstLNCConnDeadline: 10 * time.Minute,
```
Comment (guggero) [inline config.go:322]: nit: extract into constant?

File: session_rpcserver.go [hunk 4]
```diff
@@ -214,15 +215,92 @@ func (s *sessionRpcServer) resumeSession(sess *session.Session) error {
...
+			}
+		}
+
+		// Start the deadline timer.
+		firstConnectionDeadline = time.NewTimer(
```
Comment (guggero) [inline session_rpcserver.go:246]: I think we could use `time.AfterFunc()` here that closes the `firstConnTimeout`. Then in the `onNewStatus` callback we only stop that timer and we don't need a whole other goroutine (the debug comment could be moved into the `onNewStatus` callback).

**Review by guggero — COMMENTED** [review]
Nice, looks pretty good and should be a great security improvement!
I left a suggestion for making the logic a bit easier to understand.

File: session_rpcserver.go (hunk 2 above)
Comment (ellemouton) [reply session_rpcserver.go]: erg yeah sorry 🙈 also, there should be a `return` after this 🙈

File: session_rpcserver.go (hunk 4 above)
Comment (ellemouton) [reply session_rpcserver.go]: omg yeeeesss!!! thank you :) updated!

**Review by ellemouton — COMMENTED** [review]
thanks @guggero ! updated as per you suggestion 🚀  much neater the way you have suggested :)

**Comment by ellemouton** [comment]
hmmm the itest keeps failing....i must have broken something on the latest update.  looking into it

**Comment by ellemouton** [comment]
hmmmm having the same flake on master actually.....

**Comment by ellemouton** [comment]
hahaha found the issue 🙈 when I extracted the timeout into a variable, I used seconds instead of minutes 🙈  should be g2g now! I also added another commit that adds LNC client logging to the itests just to make it easier to debug these things in future

**Review by guggero — APPROVED** [review]
utACK, LGTM :tada:

**Review by jamaljsr — APPROVED** [review]
tACK 🔥

The deadline works as advertised. Thanks for this great update.

**Pull request merged**